x-pack/plugins/ml/index.js

```diff
@@ -44,6 +44,12 @@ export const ml = (kibana) => {
       styleSheetPaths: `${__dirname}/public/index.scss`,
       hacks: ['plugins/ml/hacks/toggle_app_link_in_nav'],
       home: ['plugins/ml/register_feature'],
+      injectDefaultVars(server) {
+        const config = server.config();
+        return {
+          mlEnabled: config.get('xpack.ml.enabled'),
```

Comment:
Is this just looking for the `xpack.ml.enabled` setting in kibana.yml? If so, using this as the check as to whether to add the 'Upload data from log file' link to the Home page LGTM. 

I believe APM have been using this `mlEnabled` flag to check whether to add a link to creating ML jobs from APM dashboards, but that is being edited in #24486 as they now need to check the type of license, and not just whether the ML feature is enabled.